common/analysis/lint_rule_status.cc

```diff
@@ -82,11 +82,33 @@ void LintStatusFormatter::FormatLintRuleStatuses(
       violations.insert(LintViolationWithStatus(&violation, &status));
     }
   }
+  return violations;
+}
+
+void LintStatusFormatter::FormatLintRuleStatuses(
+    std::ostream* stream, const std::vector<LintRuleStatus>& statuses,
+    absl::string_view base, absl::string_view path) const {
+  auto violations = PrepareViolations(statuses);
+  for (auto violation : violations) {
+    FormatViolation(stream, *violation.violation, base, path,
+                    violation.status->url, violation.status->lint_rule_name);
+    *stream << std::endl;
+  }
+}
+
+void LintStatusFormatter::FormatLintRuleStatuses(
+    std::ostream* stream, const std::vector<LintRuleStatus>& statuses,
+    absl::string_view base, absl::string_view path,
+    const std::vector<absl::string_view>& lines) const {
+  auto violations = PrepareViolations(statuses);
 
   for (auto violation : violations) {
     FormatViolation(stream, *violation.violation, base, path,
                     violation.status->url, violation.status->lint_rule_name);
     *stream << std::endl;
+    auto cursor = line_column_map_(violation.violation->token.left(base));
```

Comment:
... execute this `if (lines.size() > cursor)`.
that way, we can deliberately choose to just pass `{}` in places in which we don't want to show context.

(also slightly more robust in case we have `cursor.line` for whatever reason is outside `lines[]`

The added benefit is, that the compiler will find us all the places where the old function is used, so that we can make a decision at all of these places if we want context or not.